stdlib/LinearAlgebra/src/svd.jl

```diff
@@ -98,7 +98,9 @@ function svd!(A::StridedMatrix{T}; full::Bool = false, alg::Algorithm = default_
     end
     SVD(u,s,vt)
 end
-
+function svd!(A::StridedVector{T}; full::Bool = false, alg::Algorithm = default_svd_alg(A)) where T<:BlasFloat
+    svd!(reshape(A, (length(A), 1)), full = full, alg = alg)
```

Comment:
Couldn't you just return the result, since there is an analytical formula, rather than reshaping?

Reply:
I wonder if calling `svd!` with

`svd!(view(A, :, 1:1), ...)`

would do the same job, avoiding thus reshaping?

Reply:
`view`ing with a different number of indices than there are dimensions does a reshape internally.

Reply:
I don't think the `reshape` was @stevengj's issue, rather the fact that I call a function instead of writing out the result manually. Writing out the result is easy for the default `full = false`, but for `full = true` we need the left square unitary matrix. This is what I could come up with so far (which is super-ugly!):
```julia
function mysvd!(A::StridedVector{T}; full::Bool = false, alg::Algorithm = default_svd_alg(A)) where T
    if !full
        normA = norm(A)
        normalize!(A)
        return SVD(A, [normA], [one(T)])
    else
        F = qr!(reshape(A, (length(A), 1))) # need to reshape, no qr!(::Vector)
        return SVD(F.Q*Matrix(I, length(A), length(A)), [first(F.R)], fill!(Matrix{T}(undef, 1, 1), 1))
    end
end
```
I'm not sure this is worth the hassle, but maybe we find better ways to code it.

Reply:
The (default) non-full case seems to be really worth it writing out the result directly, for the other one the explicit QR approach takes longer and requires more allocations.